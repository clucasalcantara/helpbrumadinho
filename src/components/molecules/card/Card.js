import React from 'react'
import { Link } from 'react-router-dom'

const Card = ({ data: { id, name, image, location, accepts } }) => (
  <div key={id} className="column">
    <Link className="ui card" to={`location/${id}`}>
      <div className="content">
        <img
          src={image}
          alt={`${name}, ponto de coleta helpbrumadinho`}
          className="ui mini right floated image"
        />
        <div className="header">{name}</div>
        <div className="meta">{location}</div>
        <div className="description">{accepts}</div>
      </div>
    </Link>
  </div>
)

export default Card
