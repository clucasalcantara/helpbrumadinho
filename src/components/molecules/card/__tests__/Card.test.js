import React from 'react'
import { shallow } from 'enzyme'
import Card from '../Card'

describe('Card component', () => {
  it('should renders without crash', () => {
    const cardData = {
      name: 'Seven Wonders',
      image: 'x'
    }
    const wrapper = shallow(<Card data={cardData} />)

    expect(wrapper).toMatchSnapshot()
  })
})
