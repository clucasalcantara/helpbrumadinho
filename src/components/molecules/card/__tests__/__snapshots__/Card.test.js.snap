// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should renders without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
      Object {
        "image": "x",
        "name": "Seven Wonders",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        className="ui card"
        replace={false}
        to="location/undefined"
      >
        <div
          className="content"
        >
          <img
            alt="Seven Wonders, ponto de coleta helpbrumadinho"
            className="ui mini right floated image"
            src="x"
          />
          <div
            className="header"
          >
            Seven Wonders
          </div>
          <div
            className="meta"
          />
          <div
            className="description"
          />
        </div>
      </Link>,
      "className": "column",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="content"
        >
          <img
            alt="Seven Wonders, ponto de coleta helpbrumadinho"
            className="ui mini right floated image"
            src="x"
          />
          <div
            className="header"
          >
            Seven Wonders
          </div>
          <div
            className="meta"
          />
          <div
            className="description"
          />
        </div>,
        "className": "ui card",
        "replace": false,
        "to": "location/undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Seven Wonders, ponto de coleta helpbrumadinho"
              className="ui mini right floated image"
              src="x"
            />,
            <div
              className="header"
            >
              Seven Wonders
            </div>,
            <div
              className="meta"
            />,
            <div
              className="description"
            />,
          ],
          "className": "content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Seven Wonders, ponto de coleta helpbrumadinho",
              "className": "ui mini right floated image",
              "src": "x",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Seven Wonders",
              "className": "header",
            },
            "ref": null,
            "rendered": "Seven Wonders",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "meta",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "description",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="ui card"
          replace={false}
          to="location/undefined"
        >
          <div
            className="content"
          >
            <img
              alt="Seven Wonders, ponto de coleta helpbrumadinho"
              className="ui mini right floated image"
              src="x"
            />
            <div
              className="header"
            >
              Seven Wonders
            </div>
            <div
              className="meta"
            />
            <div
              className="description"
            />
          </div>
        </Link>,
        "className": "column",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="content"
          >
            <img
              alt="Seven Wonders, ponto de coleta helpbrumadinho"
              className="ui mini right floated image"
              src="x"
            />
            <div
              className="header"
            >
              Seven Wonders
            </div>
            <div
              className="meta"
            />
            <div
              className="description"
            />
          </div>,
          "className": "ui card",
          "replace": false,
          "to": "location/undefined",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Seven Wonders, ponto de coleta helpbrumadinho"
                className="ui mini right floated image"
                src="x"
              />,
              <div
                className="header"
              >
                Seven Wonders
              </div>,
              <div
                className="meta"
              />,
              <div
                className="description"
              />,
            ],
            "className": "content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Seven Wonders, ponto de coleta helpbrumadinho",
                "className": "ui mini right floated image",
                "src": "x",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Seven Wonders",
                "className": "header",
              },
              "ref": null,
              "rendered": "Seven Wonders",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "meta",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "description",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
